engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EqualsComplexNode.java

```diff
@@ -543,6 +543,12 @@ boolean isObjectWithMembers(Object object, InteropLibrary interop) {
     if (interop.isTime(object)) {
       return false;
     }
+    if (interop.isDuration(object)) {
```

Comment:
Both TimeZone and Duration now have members, so we need to add these guards here.

Reply:
See my answer in https://github.com/enso-org/enso/pull/11687#issuecomment-2528845368 - yes it is a mess.